src/modules/tracker.js

```diff
@@ -1228,6 +1229,401 @@ class Tracker {
     return new Error('A parameters object with an "item_id" property is required.');
   }
 
+  /**
+   * Send quiz results loaded event to API
+   *
+   * @function trackQuizResultsLoaded
+   * @param {object} parameters - Additional parameters to be sent with request
+   * @param {string} parameters.quizId - Quiz Id
+   * @param {string} parameters.quizVersionId - Quiz version Id
+   * @param {string} parameters.url - Current page url
+   * @param {string} [parameters.section='Products'] - Index section
+   * @param {number} [parameters.resultCount] - Total number of results
+   * @param {number} [parameters.resultPage] - The page of the results
+   * @param {string} [parameters.resultId] - Quiz result identifier (returned in response from Constructor)
+   * @param {object} [networkParameters] - Parameters relevant to the network request
+   * @param {number} [networkParameters.timeout] - Request timeout (in milliseconds)
+   * @returns {(true|Error)}
+   * @description User viewed a quiz results page
+   * @example
+   * constructorio.tracker.trackQuizResultsLoaded(
+   *     {
+   *         quizId: 'coffee-quiz',
+   *         quizVersionId: '1231244'
+   *         url: www.example.com
+   *         resultCount: 167,
+   *     },
+   * );
+   */
+  trackQuizResultsLoaded(parameters, networkParameters = {}) {
+    // Ensure parameters are provided (required)
+    if (parameters && typeof parameters === 'object' && !Array.isArray(parameters)) {
+      const requestPath = `${this.options.serviceUrl}/v2/behavioral_action/quiz_result_load?`;
+      const { quizId, quizVersionId, url, section, resultCount, resultId, resultPage } = parameters;
+      const queryParams = {};
+      const bodyParams = {};
+
+      if (typeof quizId !== 'string') {
+        return new Error('"quizId" is a required parameter of type string');
+      }
+
+      if (typeof quizVersionId !== 'string') {
+        return new Error('"quizVersionId" is a required parameter of type string');
+      }
+
+      if (typeof url !== 'string') {
+        return new Error('"url" is a required parameter of type string');
+      }
+
+      bodyParams.quizId = quizId;
+      bodyParams.quizVersionId = quizVersionId;
+      bodyParams.url = url;
+
+      if (!helpers.isNil(section)) {
+        if (typeof section !== 'string') {
+          return new Error('"section" must be a string');
+        }
+        queryParams.section = section;
+        bodyParams.section = section;
+      }
+
+      if (!helpers.isNil(resultCount)) {
```

Comment:
Do we need both isNil and typeof check?

Since typeof null/undefined won't be string/number so I don't think we need the nil check.